src/classes/HH_HouseholdNamingSettingValidator_TEST.cls

```diff
@@ -0,0 +1,361 @@
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+ 
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT 
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS 
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE 
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, 
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, 
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; 
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER 
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT 
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN 
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2017
+* @group Households
+* @description Unit tests for HH_HouseholdNamingSettingValidator class
+*/
+@isTest
+private class HH_HouseholdNamingSettingValidator_TEST {
+    /*********************************************************************************************************
+    * @description Constant containing invalid Household Name Format 
+    */
+    private static final String INVALID_HHNAME_FORMAT = 'The {!LastName}, {!{!FirstName}} {!npe01__WorkEmail__c}} Family';
+    /*********************************************************************************************************
+    * @description Constant containing invalid Formal Greeting Format 
+    */
+    private static final String INVALID_FORMAL_GREETING_FORMAT = '{!{!Title} {!FirstName}}}{!LastName}';
+    /*********************************************************************************************************
+    * @description Constant containing invalid Informal Greeting Format 
+    */
+    private static final String INVALID_INFORMAL_GREETING_FORMAT = '{!{!FirstName}}}';
+
+    /*********************************************************************************************************
+    * @description Notification used to store and verify errors (if any) reported during validation
+    */
+    private static HH_HouseholdNamingSettingValidator.Notification notification;
+    /*********************************************************************************************************
+    * @description Validator used to validate Household Naming Settings
+    */
+    private static HH_HouseholdNamingSettingValidator validator;
+
+
+    /*********************************************************************************************************
+    * @description Static method to initialize variables before each unit test
+    */
+    static {
```

Comment:
most of our test code was written before @testsetup was introduced by Salesforce, and most of us just follow patterns we've used in the past.  I'm fine with people using testsetup, or init routines, or just initializing static vars in tests.  I'm just not familiar with creating an anonymous static block to init static vars.  Why not just put the initialization on the same line as the declaration, if you don't need a full init routine?